client/app/hosting/framework/envvar/add/hosting-framework-envvar-add.html

```diff
@@ -0,0 +1,59 @@
+<div data-ng-controller="controllers.Hosting.Framework.Envvar.create as ctrl">
+    <div data-wizard
+         data-wizard-on-cancel="resetAction"
+         data-wizard-on-finish="create"
+         data-wizard-title="i18n.hosting_tab_FRAMEWORK_envvar_create_title">
+
+        <div data-wizard-step
+             data-wizard-step-valid="isValid()">
+
+            <form class="form-horizontal" name="addEnvvarForm" data-ng-if="!ctrl.loading">
+                <p>
+                    <small class="text-danger">*</small>
+                    <small data-i18n-static="required_fields"></small>
+                </p>
+
+                <div class="form-group"
+                     data-ng-class="{ 'has-error': ctrl.entryToCreate.key && !ctrl.isKeyValid() }">
```

Comment:
You should use the AngularJS way of handling errors:
* https://www.w3schools.com/angular/angular_validation.asp
* https://scotch.io/tutorials/angularjs-form-validation